### PR TITLE
Make port use an integer value, not implicit string conversion

### PR DESCRIPTION
**Pull request by deanwilson**
This is an early step to support future/parser puppet 4. Our rspec
tests currently specify (correctly) that port is an int. Our manifest
should reflect that choice of type.

We also shouldn't be running our apps on priviledged ports so we set
the lowest allowed port to be 1025.

This will require apps using strings to be converted over to ints
when we start to run puppet 4. Those will be done in a seperate PR.

### CODE REVIEW

**Comment by deanwilson** [comment]
if the validate_integer was moved in to the if would this still cover all cases where we want to use port?

**Comment by rjw1** [comment]
I wonder if the port should be mandatory?

**Comment by rjw1** [comment]
:+1:
 The intent is correct. 
I do wonder if the fix should go in here as well so the PR doesnt break deploys. 

What would happen if we set the default to null.

File: modules/govuk/manifests/app.pp [hunk 1]
```diff
@@ -269,14 +269,20 @@
   if ! ($app_type in ['procfile', 'rack', 'bare']) {
     fail 'Invalid argument $app_type to govuk::app! Must be one of "procfile", "rack", or "bare"'
   }
-  if ($app_type in ['procfile', 'rack']) and ($port == 'NOTSET') {
+
+  if ($app_type in ['procfile', 'rack']) and ($port == 0) {
```
Comment (mattbostock) [inline modules/govuk/manifests/app.pp]: Just realised that this conditional is redundant as any ports outside of 1025-65535 will be caught by `validate_integer` below.

Comment (rjw1) [inline modules/govuk/manifests/app.pp]: It is a more useful error message though.

**Pull request merged**